spec/api.md

```diff
@@ -0,0 +1,129 @@
+# CAS API Documentation
+
+This document describes the HTTP API endpoints used by the CAS (Content Addressable Storage) client to interact with the remote CAS server.
+
+## Authentication
+
+- TODO
+
+## Endpoints
+
+### 1. Get File Reconstruction
+
+**Description**: Retrieves reconstruction information for a specific file, optionally with byte range support.
+
+- **Path**: `/v1/reconstruction/{file_id}`
+- **Method**: `GET`
+- **Parameters**:
+  - `file_id`: MerkleHash in hex format
+- **Headers**:
+  - `Range`: Optional. Format: `bytes={start}-{end}` (inclusive end)
+- **Body**: None
+- **Response**: JSON (`QueryReconstructionResponse`)
+
+  ```json
+  {
+    "offset_into_first_range": 0,
+    "terms": [...],
+    "fetch_info": {...}
+  }
+  ```
+
+- **Error Responses**:
+  - 400 bad request
+  - 404 File not found
+  - 416 Range Not Satisfiable: When requested byte range start exceeds the end of a file
+
+### 2. Batch Get Reconstruction
+
+**Description**: Retrieves reconstruction information for multiple files in a single request.
+
+- **Path**: `/v1/reconstructions`
```

Comment:
The convention for a RESTful API is to have both the singular and batch versions of the api using the same path. 

In other words, the above API should be `/v1/reconstructions/{id}` and this would be `/v1/reconstructions`.